### DISCUSSION TITLE
browsr.reloadSession error in WebdriverIO v7.16.7

### DISCUSSION THREAD

**Discussion by tzurp** [webdriverio/webdriverio#7701] [Community]
I put `browser.reloadSession()` inside `afterEach` mocha hook and run several tests in a file. Once in several tests I get this error: "ERROR webdriver: Request failed with status 404 due to invalid session id: invalid session id". Seems like a bug to me because our master branch has wdio v7.14.1 and I never encountered this error. Can anyone confirm this?

**Comment by christian-bromann** [comment]
@tzurp we didn't made any changes to the command or underlying workings. Things should work the same. Can you compare logs?

**Reply by tzurp** [reply]
@christian-bromann I think I found the source of the problem. It is related to the number of concurrent sessions we can run in v7.16.x compared to v7.14.1. In our project we are are using `maxInstances: 5`. When calling `browser.reloadSession()` it creates another session (I guess there is a moment with 6 concurrent sessions) and the framework in our environment can handle it with no problems with v7.14.1. On the other hand some sessions are failed to be created in v7.16.x. I managed to solved the problem by reducing the max instances settings to `maxInstances: 4` and separating the suite that we use frequent `browser.reloadSession()` after each test. While I didn't measure the machine metrics, my guess is that v7.16.x might be a little bit heavier on our environment and consume more resources that can be balanced by reducing the concurrent sessions. Does it make sense to you?

**Reply by christian-bromann** [reply]
> Does it make sense to you?

Yes, not sure why it would be heavier on the environment though.

**Reply by NaZaRKIN123** [reply]
Having the same issue here with 7.32.1 + Cucumber and `browser.reloadSession()` inside `afterEach`.
